### PR TITLE
Added exception for division by zero

### PR DESCRIPTION
**Pull request by nishnik**
Related to #821

### CODE REVIEW

**Comment by certik** [comment]
Thanks! +1 to merge if tests pass.

**Comment by nishnik** [comment]
Travis is failing because we have used:
`r1 = asec(integer(0));` in `TEST_CASE("Asec: eval_arb", "[eval_arb]")`
and
`r1 = acsc(integer(0));` in `TEST_CASE("ACsc: eval_arb", "[eval_arb]")`
Which are indeterminate, and are mentioned in comments too.
What is needed now ? @certik

**Comment by nishnik** [comment]
Ping @isuruf @certik

**Comment by CodeMaxx** [comment]
@nishnik These tests will have to be removed in that case since the purpose of the this PR is to remove indeterminates.( I'm not sure if there is any other way round this. @certik @isuruf )

Also the message is not very clear since it might not be obvious to the user how `tan(pi/2)` causes a divide by zero(except if he mentally expands it to `sin/cos` and calculates). The message should rather be **The Result is Undefined and/or Infinity** or something like that.

**Comment by nishnik** [comment]
@CodeMaxx I gave a thought over this, then I tried running `1/0` in  a python terminal and it gave the error `integer division or modulo by zero`, So rather than giving it a message like `Result is infinity` or `Result is undefined`, which at times can be confusing like `x/0` is not `infinite`, it is `indeterminate` and `lim x->0 (6/x)` is not `indeterminate`, it is `infinite` (Someone might face problem while tracing the bug) , I gave it the standard error message.

**Comment by abhinavagarwal07** [comment]
Seems fine to me. Test have also passed.

**Comment by isuruf** [comment]
Looks good to me. Can you add the example from #821 as a test case?

File: symengine/tests/basic/test_functions.cpp [hunk 1]
```diff
@@ -440,6 +440,8 @@ TEST_CASE("Tan: functions", "[functions]")
     r1 = tan(add(sub(mul(i12, pi), y), div(pi, i2)));
     r2 = cot(y);
     REQUIRE(eq(*r1, *r2));
+
+    CHECK_THROWS_AS(tan(mul(integer(5), div(pi, i2))), std::runtime_error);
```
Comment (nishnik) [inline symengine/tests/basic/test_functions.cpp]: @isuruf This replicates the test case from #821

**Comment by isuruf** [comment]
Thanks for the PR @nishnik

**Pull request merged**